core/src/main/java/org/mule/runtime/core/processor/chain/AbstractMessageProcessorChain.java

```diff
@@ -103,16 +109,55 @@ protected Event doProcess(Event event) throws MuleException {
   @Override
   public Publisher<Event> apply(Publisher<Event> publisher) {
     List<BiFunction<Processor, ReactiveProcessor, ReactiveProcessor>> interceptorsToBeExecuted = resolveInterceptors();
-
     Flux<Event> stream = from(publisher);
     for (Processor processor : getProcessorsToExecute()) {
+      stream = stream.transform(applyInterceptors(interceptorsToBeExecuted, processor));
+    }
+    return stream;
+  }
+
+  private ReactiveProcessor applyInterceptors(List<BiFunction<Processor, ReactiveProcessor, ReactiveProcessor>> interceptorsToBeExecuted,
+                                              Processor processor) {
+    return stream -> {
       ReactiveProcessor processorFunction = processor;
       for (BiFunction<Processor, ReactiveProcessor, ReactiveProcessor> interceptor : interceptorsToBeExecuted) {
         processorFunction = interceptor.apply(processor, processorFunction);
       }
-      stream = stream.transform(processorFunction);
+      if (flowConstruct instanceof Pipeline) {
+        processorFunction = ((Pipeline) flowConstruct).getProcessingStrategy()
+            .onProcessor(processorFunction);
+      }
+
+      ReactiveProcessor finalProcessorFunction = processorFunction;
+      stream = from(stream).flatMap(event -> {
+        try {
+          return Flux.just(event)
+              .transform(finalProcessorFunction)
+              .onErrorResumeWith(MessagingException.class, handleError(event.getContext()));
+        } catch (Throwable throwable) {
```

Comment:
Is this catch ever executed? If the flux is being used like a "builder" here, and not run (with force/subscribe/etc), exceptions thrown in finalProcessorFunction or handleError won't be caught here.